fusillade/api/__init__.py

```diff
@@ -0,0 +1,204 @@
+    code is never invoked, and the common_error_handler in the connexion.App object is never called.  This means error
+    messsages are not returned using our standard error formats.
+
+    The solution is to trap the validation results, and if it fails, exit the validation flow.  We catch the exception
+    at the top level where the various validators are called, and return a value according to our specs.
+    """
+
+    @staticmethod
+    def validate_parameter(*args, **kwargs):
+        result = ParameterValidator.validate_parameter(*args, **kwargs)
+        if result is not None:
+            raise FusilladeBindingException(result)
+        return result
+
+    def __call__(self, function):
+        origwrapper = super().__call__(function)
+
+        @functools.wraps(origwrapper)
+        def wrapper(request):
+            try:
+                return origwrapper(request)
+            except FusilladeBindingException as ex:
+                return ex.to_problem()
+
+        return wrapper
+
+
+class FusilladeRequestBodyValidator(RequestBodyValidator):
+    """
+    The RequestBodyValidator provided by Connexion immediately returns a value if the validation fails.  Therefore, our
+    code is never invoked, and the common_error_handler in the connexion.App object is never called.  This means error
+    messsages are not returned using our standard error formats.
+
+    The solution is to trap the validation results, and if it fails, exit the validation flow.  We catch the exception
+    at the top level where the various validators are called.
+    """
+
+    def validate_schema(self, *args, **kwargs):
+        result = super().validate_schema(*args, **kwargs)
+        if result is not None:
+            raise FusilladeBindingException(result.body['detail'])
+        return result
+
+    def __call__(self, function):
+        origwrapper = super().__call__(function)
+
+        @functools.wraps(origwrapper)
+        def wrapper(request):
+            try:
+                return origwrapper(request)
+            except FusilladeBindingException as ex:
+                return ex.to_problem()
+
+        return wrapper
+
+
+class ChaliceWithConnexion(chalice.Chalice):
+    """
+    Subclasses Chalice to host a Connexion app, route and proxy requests to it.
+    """
+    def __init__(self, swagger_spec_path, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.swagger_spec_path = swagger_spec_path
+        self.connexion_app = self.create_connexion_app()
+        self.connexion_full_dispatch_request = self.connexion_app.app.full_dispatch_request
+        self.connexion_request_context = self.connexion_app.app.test_request_context
+        self.trailing_slash_routes = []
+        routes = collections.defaultdict(list)
+        for rule in self.connexion_app.app.url_map.iter_rules():
+            route = re.sub(r"<(.+?)(:.+?)?>", r"{\1}", rule.rule)
+            if route.endswith("/"):
```

Comment:
`if route.endswith("/"):` seems unnecessary.  `rstrip("/")` will ignore the end if there's no slash there.

Reply:
will fix

Reply:
this conditionally stores the value in another locations. This is needed to build the routes